lib/grpc/message.ex

```diff
@@ -172,4 +172,28 @@ defmodule GRPC.Message do
   def get_message(_) do
     false
   end
+
+  def get_message(data, nil = _compressor) do
+    case data do
+      <<flag::unsigned-integer-size(8), length::unsigned-integer-size(32),
+        message::bytes-size(length), rest::binary>> ->
+        {{flag, message}, rest}
+
+      _other ->
+        data
+    end
+  end
+
+  def get_message(data, compressor) do
+    case data do
+      <<1::8, length::unsigned-integer-32, message::bytes-size(length), rest::binary>> ->
+        {{1, compressor.decompress(message)}, rest}
+
+      <<0::8, length::unsigned-integer-32, message::bytes-size(length), rest::binary>> ->
+        {{0, message}, rest}
```

Comment:
just to make sure, we don't need to decompress this message here, right? It's what the first bit is saying

Reply:
yup! that's what the first byte is about. I've looked at this doc for ref: https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md